core-feature-pack/src/main/resources/content/bin/domain.bat

```diff
@@ -78,6 +78,43 @@ if "%OS%" == "Windows_NT" (
   set "PROGNAME=domain.bat"
 )
 
+setlocal EnableDelayedExpansion
+rem Add -server to the JVM options, if supported by JDK
+set JVM_OPTVERSION="-version"
+echo "%JAVA_OPTS%" | findstr /I \-d64 > nul
+if errorlevel == 0 (
+    set JVM_OPTVERSION="-d64 -version"
+    echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"Unrecognized option" > nul
+     if errorlevel == 0 (
+            set JVM_OPTVERSION="-version"
+      )
+) else (
+    echo "%JAVA_OPTS%" | findstr /I \-d32 > nul
+    if errorlevel == 0 (
+        set JVM_OPTVERSION="-d32 -version"
+        echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"Unrecognized option" > nul
+         if errorlevel == 0 (
+                set JVM_OPTVERSION="-version"
+          )
+    )
+)
+
+echo "%JAVA_OPTS%" | findstr /I \-server > nul
+if errorlevel == 1 (
+
+  echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"HotSpot" > nul
```

Comment:
HotSpot detection is currently on line 103 (https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/src/main/resources/content/bin/domain.bat#L103), this `-server` settings is used on HC, PC and servers. Is this code duplication?

Reply:
Lines 84-100 are checking for text "-d64" and "-d32".
  Lines 103-114 are checking for "HotSpot" or "OpenJDK"
  There seems to be some combination of -d64 HotSpot, -d32 HotSpot,
  -d64 OpenJDK, -d32 OpenJdk that must be identified on all platforms.

Reply:
> Lines 103-114 are checking for "HotSpot" or "OpenJDK"

Ok ... so are you agree with me, that this hotspot detection is a code duplication? 

See [original line 103](https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/src/main/resources/content/bin/domain.bat#L103):

```
 "%JAVA%" -server -version 2>&1 | findstr /I hotspot > nul
if not errorlevel == 1 (
...
```

and compare it with your new code:

```
 echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"HotSpot" > nul
  if errorlevel == 0 (
```

Reply:
Btw, is the "echo" command really needed in your code?

Reply:
@jamezp Can you please review this PR as well?

Reply:
@marekkopecky I'll see if I can find the time. I'm stretched pretty thin ATM :)

Reply:
Yes it is.   This how var contents is pass to the function findstr.

Reply:
The task (lines 140-144) is performing is NOT a duplicate or (lines 102-114)
At line 102 if the user does not have "-server" set in JAVA_OPTS then 
that if block is executed.   When "-server" is already set in JAVA_OPTS
the "if" block (lines 103-114) is skipped.  At (lines 140-144) we must 
determine if "hotspot" is set and set the 2 variables accordingly.  This 
block of code is not a duplicate needs to remain as is.